src/api/db/data/20250404105011_change_decision_report_subscriptions.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class ChangeDecisionReportSubscriptions < ActiveRecord::Migration[7.0]
+  def up
+    report_subscriptions = EventSubscription.where(eventtype: %w[Event::ReportForComment Event::ReportForPackage Event::ReportForProject Event::ReportForRequest Event::ReportForUser])
+    report_subscriptions.map { |s| s.attributes.slice('user_id', 'group_id', 'receiver_role', 'channel') }.uniq.each do |report|
+      EventSubscription.create(report.merge({ eventtype: 'Event::Report' }))
```

Comment:
```suggestion
    report_subscriptions = EventSubscription.where(enabled: true, eventtype: %w[Event::ReportForComment Event::ReportForPackage Event::ReportForProject Event::ReportForRequest Event::ReportForUser])
    report_subscriptions.map { |s| s.attributes.slice('enabled', 'user_id', 'group_id', 'receiver_role', 'channel') }.uniq.each do |report|
      EventSubscription.create(report.merge({ eventtype: 'Event::Report' }))
```
If we don't filter for `enabled: true`, we will create a lot of `enabled: false` records. Or am I missing something?

Same thing in the `decision_subscriptions...` loop below.

Reply:
We might as well migrate the actual enabled state too, so I omitted `enabled: true`, but included it in the slice

Reply:
If we don't filter by `enabled: true` and include `enabled` in the slice, we could end with unwanted results, I think. For example, a user has web subscriptions for:
1. `eventtype: Event::ReportForComment, enabled: true`
1. `eventtype: Event::ReportForPackage, enabled: false`

`report_subscriptions.map { |s| s.attributes.slice('enabled', 'user_id', 'group_id', 'receiver_role', 'channel') }.uniq` would return one record with `enabled: true` and one record with `enabled: false`.

Does this make sense?

Reply:
Yeah, though this shouldn't be a problem since there doesn't seem to be any constraint that would stop us from having multiple records with different values

Reply:
I changed the code, feel free to take a look again

Reply:
Thanks for the fix. :+1: